packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -84,7 +88,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     throw new Error(`Failed to find text node. Node: ${node.textContent}, offset: ${offset}`);
   }
 
-  const textOffset = Math.max(0, offset - runningOffset);
+  let textOffset = Math.max(0, offset - runningOffset);
+
+  const originalText = nodeElement.getAttribute('original-text');
+
+  // If attribute 'original-text' is present then
+  // the string contains some encoded entities
+  // so, we need adjust the text offset
+  if (originalText) {
+    // To properly calculate the offset of the string we want to highlight
+    // we need to get the offset from the decoded html text and subtract
+    // it from the original encoded html offset
```

Comment:
Update this comment to specify that the offsets are based on the HTML string, which may contain HTML entities; but need to calculate the offset in the DOM, where the HTML entities have been rendered, as single characters now. Maybe throw in an example of `&amp;` in the HTML string vs `&` in the DOM.